### DISCUSSION TITLE
VS Code vs Webstorm

### DISCUSSION THREAD

**Discussion by renangasperi** [frontendbr/forum#1687] [Perguntas e Respostas]
Eu sou um usuário do VS Code desde quando comecei a programar e, ultimamente, tenho visto algumas pessoas utilizando o Webstorm(JetBrains) e gostando bastante. Já vi alguns posts de blogs sobre o assunto e vou deixar embaixo para quem tiver interesse.
Mas o que vocês acham? Já tentaram ou tem interesse em migrar pro Webstorm?

Links que encontrei:
https://dev.to/mokkapps/why-i-switched-from-visual-studio-code-to-jetbrains-webstorm-939
https://www.slant.co/versus/5980/5982/~webstorm_vs_visual-studio-code

**Comment by italosilva01-zz** [comment]
Olha. Eu acredito que o vscode sai ganhando. Tudo bem que vscode é só um editor de texto e Webstorm é uma IDE, porém o VS code tem a proposta de evoluir junto com o programador. O VS code faz isso por meio de plugins. Assim, uma hora ou outra o VS Code irá fazer a mesma coisa que uma IDE robusta faz, se tornando uma IDE customizada.

**Comment by oHervis** [comment]
Concordo com o @italosilva01, vscode pode evoluir com você, ou seja, você só vai instalar aquilo que realmente vai usar, não que o webstorm seja ruim, mas pode ser uma ferramenta com muita coisa que você pode nunca usar.

**Comment by wilsonneto-swe** [comment]
Eu já usei por pouco tempo, mas gostei muito do VS Code, a customização, consigo usar para todas as linguagens que trabalho e tal... Minha preferência é VS Code frente a qualquer editor de textos e IDEs...

**Comment by eliseumds** [comment]
O VSCode não é um simples editor de texto. Tem integração com Git (plugin GitLens), debugging, refactoring, formatting (Prettier), terminal e code analysis pode ser feita com ESLint e outros plugins. Eu já usei os dois e voltei pro VSCode.

WebStorm vem com muito mais ferramentas instaladas por padrão mas você paga o preço em lentidão.

**Comment by mustafahamandosh** [comment]
sem duivida Webstorm é melhor.

**Comment by wilsonneto-swe** [comment]
sem dúvidas webstorm é mais lento...

Brincadeiras a parte, não existe o melhor editor assim como não existe a melhor ferramenta, tudo depende de em qual você se sente melhor e mais produtivo, para mim VS Code, mas ambos muitos bons, teste ambos.

**Comment by vitebo** [comment]
atualmente eu to utilizando o webstorm, mas antes eu utilizada o vscode, então vou deixar aqui os pontos que eu gostei e não gostei de cada um, pode ser que acabe ajudando.

Antes de utilizar o vscode eu utilizava o neovim, e tanto no vscode quanto no webstorm eu tenho um plugin pro vim, o que eu gosto bastante. Ambos os plugins são muito bons, mas eu consigo customizar muito mais o plugin do vscode. Um exemplo é quando eu to navegando no diretório de pastas do meu projeto, no vscode eu consegui remapear para me movimentar no diretório através das teclas `h-j-k-l`. No webstorm ainda não consegui fazer isso. _(caso alguém saiba uma maneira de fazer isso eu ficaria feliz)_.

Atualmente eu trabalho em um projeto que é um monólito de [elixir](https://elixir-lang.org/) com [vue](https://vuejs.org/).
Em ambos os casos eu precisei adicionar um plugin para ter o highlight da syntaxe. No webstorm foi meio chato de fazer isso, porque eu sou obrigado a adicionar o path do sdk do elixir, se não fica uma msg de erro no editor que não pra tirar. O único problema disso é porque eu rodo a aplicação inteira dentro de um container docker, então eu não precisaria ter o elixir instalado na minha maquina, mas precisei instalar só por causa do webstrom. Com o vscode eu não tive esse problema, foi muito mais simples, contudo pelo fato do webstorm ter acesso ao sdk do elixir, as sugestões do plugin são muito mais completas.

Eu senti o suporte para o vue no webstorm muito melhor do que no vscode _(eu usava o plugin do [Vetur](https://github.com/vuejs/vetur))_. No webstorm eu consigo ter autocomplete dos componentes no template, e ele ja me diz quais são as props que aquele componente aceita. Eu consigo extrair um pedaço de um componente para um novo componente apenas com uma ação. No vscode apesar deu ter um suporte bom, eu não tenho essas funcionalidades mais avançadas.

Para debugar eu gosto mais do webstorm, tanto os testes da aplicação quanto a aplicação rodando no browser tbm. Eu achei muito mais simples de fazer as coisas. No vscode tbm tem plugin pra isso mas eles nunca funcionaram muito bem cmg.

Por ultimo uma função que eu gosto muito do webstorm, é que ele consegue gerar um [diagrama mostrando todas a dependências do seu modulo](https://www.jetbrains.com/help/webstorm/module-dependency-diagram.html#js_ts_module_dependency_diagram_analyze).
Isso é muito legal para documentar, ou colocar em um PR para exemplificar melhor oque foi feito. Eu ainda não consegui encontrar nenhuma ferramenta externa que consiga fazer isso. 

Em um resumo geral, eu acho o webstorm mais completo e menos flexível, e o vscode mais flexível e customizável. Hoje eu to no webstorm porque eu to testando bastante coisa e eu tenho a licença de estudante que me garante acesso a todos os produtos da jetbrains free, mas para desenvolvimento web frontend eu não sei se eu pagaria o valor cheio nessa ferramente, sendo que eu tenho o vscode que é tão bom quanto, e é free.

Só lembrando que eu não sou nenhum tipo de expert em nenhuma dessas ferramentas, pelo contrário, eu sou só um charlinho que gosta muito de fuçar :laughing:

**Comment by mustafahamandosh** [comment]
@wilsonneto-dev haha concordo com ti não ha o melhor.
Eu já utilizei os dois, mas o webstorm é mais inteligente.
Eu defendo o webstorm versão pago e não o gratuito.
O gratuito um é ruim de mais, mas o pago é perfeito.
Eu uso a minha conta de estudante e o jetbrains me da acesso de dois anos gratuitamente.

**Comment by onlurking** [comment]
Minhas percepções após um tempo usando WebStorm vs anos usando o VSCode:

- Plugin de Vim do VS Code (VSVim) é muito superior ao IdeaVim da Jetbrains.
- Acho a integração Git do VS Code melhor, ainda mais se ele tiver o plugin GitLens instalado.
- A refatoração do Webstorm põe o VS Code no chinelo, renomear variáveis/métodos/arquivos, extrair código pra componentes é infinitamente superior (mesmo usando os plugins Abracadabra e JavaScript Booster no VS).
- A IDE da Jetbrains é absurdamente mais feature complete, uma das features legais é a "Module Dependency Diagrams﻿" que o @vitebo mencionou e outra meio obscura que tenho usado é o "Code Coverage mode": https://www.jetbrains.com/help/webstorm/capturing-unused-javascript-code.html - que mapeia todo o código que está sendo rodado do seu projeto, muito útil para achar código morto.
- VS Code é muito mais customizável, os temas da Jetbrains inclusive são bem zoados em comparação.
- Falando de análise estática, o VS Code é muito bom com eslint, stylelint e prettier configurados, mas o Webstorm não apenas integra com eles automaticamente, como ele possui uma dezenas de novos checks, mesmo em um projeto que tem 0 erros de lint no VS Code o Webstorm deve sugerir mais fixes no codebase que ainda não haviam sido descobertos antes.

Atualmente estou usando o VS Code como editor principal, mas sempre que preciso fazer uma refatoração pesada ou antes de enviar minhas PRs pra code review eu abro o código no Webstorm, tem funcionado muito bem pra mim dessa forma.

**Comment by igorlvicente** [comment]
Não uso o WebStorm mas uso o RubyMine.
Há uns 3 anos eu mudei para o RubyMine. Sempre tivera preconceito com IDEs, mas fui meio que obrigado a passar a usar.
O que eu posso dizer é que, na minha opinião, depende do que eu quero.
Se eu quero sentir que estou botando a mão na massa e entender tudo o que está acontecendo, VSCode. Agora se o que importa é entregar valor, vou de WebStorm.

**Reply by akinors** [reply]
The Webstorm it's so old school 😉

**Reply by ojvribeiro** [reply]
> Agora se o que importa é entregar valor, vou de WebStorm.

Defina "entregar valor".

**Reply by igorlvicente** [reply]
> > Agora se o que importa é entregar valor, vou de WebStorm.
> 
> Defina "entregar valor".

Eficiência e eficácia ao executar o desenvolvimento do produto que você está trabalhando. O famoso "get shit done".
